lib/diagnostic.js

```diff
@@ -5,30 +5,60 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
-
+    let result = arguments[0];
+    for (let i = 0; i < arguments.length; i++)
+    {
+        if (result > arguments[i])
+        {
+          result = arguments[i];
+        }
+    }
+    return result;
 };
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum() {
-
+  let result = 0;
+  for (let i = 0; i < arguments.length; i++)
+  {
+      result = result + arguments[i];
+  }
+  return result;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
-const newArray = function newArray(length, defaultValue) {
-
-};
+// const newArray = function newArray(length, defaultValue) {
+//   let result = [];
+//   for (let i = 0; i < length; i++)
+//   {
+//     result[i] = defaultValue;
+//   }
+//   return result;
+// };
```

Comment:
This is where you have to assign `const newArray` and question 4 you have to use the variable in a different way.